weave/weave_client.py

```diff
@@ -805,6 +813,9 @@ def _save_nested_objects(self, obj: Any, name: Optional[str] = None) -> Any:
                 self._save_nested_objects(v)
         elif isinstance(obj, Op):
             self._save_op(obj)
+        # TODO: Kinda hacky way to dispatching Dataset with rows: Table
+        elif isinstance(obj, WeaveObject) and hasattr(obj, "rows"):
```

Comment:
this might be more robust if we iterated through the fields and looked for table properties?